src/Compilers/CSharp/Portable/Compilation/AttributeSemanticModel.cs

```diff
@@ -38,7 +41,7 @@ internal sealed class AttributeSemanticModel : MemberSemanticModel
         public static AttributeSemanticModel Create(SyntaxTreeSemanticModel containingSemanticModel, AttributeSyntax syntax, NamedTypeSymbol attributeType, AliasSymbol aliasOpt, Symbol? attributeTarget, Binder rootBinder, ImmutableDictionary<Symbol, Symbol> parentRemappedSymbolsOpt)
```

Comment:
@jcouv I'm confused by the callsite of this method. The `attributeTarget` comes from:

https://github.com/dotnet/roslyn/blob/4518b4087d086319310b9dfe055906770e1eba7d/src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs#L1288-L1302

Attribute target should be whatever symbol the attribute was placed on (types and assemblies included). But the code referenced above checks for a limited set of declarations (types and assembly attributes not included).

For semantic model, I couldn't find a scenario where we care. If this is true, it would be good to follow up with a better terminology or simplify the above code to use `GetDeclaredSymbolForNode` which will work for any syntax node.

Let me know what do you think?

Reply:
Yeah. https://github.com/dotnet/roslyn/pull/62137#discussion_r906871844